src/wp-includes/nav-menu.php

```diff
@@ -693,24 +693,28 @@ function wp_get_nav_menu_items( $menu, $args = array() ) {
 
 	static $fetched = array();
 
-	$items = get_objects_in_term( $menu->term_id, 'nav_menu' );
-	if ( is_wp_error( $items ) ) {
+	if ( ! taxonomy_exists( 'nav_menu' ) ) {
 		return false;
 	}
 
-	$defaults        = array(
+	$defaults = array(
 		'order'       => 'ASC',
 		'orderby'     => 'menu_order',
 		'post_type'   => 'nav_menu_item',
 		'post_status' => 'publish',
 		'output'      => ARRAY_A,
 		'output_key'  => 'menu_order',
 		'nopaging'    => true,
+		'tax_query'   => array(
+			array(
+				'taxonomy' => 'nav_menu',
+				'field'    => 'term_taxonomy_id',
+				'terms'    => $menu->term_taxonomy_id,
+			),
```

Comment:
This might have an adverse performance affect:

* it adds a `JOIN` to the query
* on a site with multiple menus (eg, header and footer) an additional uncached query will be made for each menu. Without the taxonomy portion of the query, the second and subsequent query will hit an existing cache.

A history lesson here might be helpful.

Reply:
For reference this change remove one query and replaces it with a very simple JOIN. See the before and after. 
### Before
<img width="1227" alt="Screenshot 2022-03-22 at 12 17 58" src="https://user-images.githubusercontent.com/237508/159480542-b3c5f2f4-4ae5-47ea-82df-4d7fc055389b.png">

### After
<img width="1255" alt="Screenshot 2022-03-22 at 12 17 45" src="https://user-images.githubusercontent.com/237508/159480647-4ff33d22-7cbd-4f45-a2c5-f61618fe3bf5.png">

Reply:
I set up a plugin to do [some rudimentary testing](https://gist.github.com/peterwilsoncc/b47ebac2dd10cfba148b326a7f1d3a60). The numbers are only useful as relative indications.

Set up:
* Vagrant local dev environment
* MySQL on same host
* Small menu: 5 posts, 5 tags
* Med menu: 13 posts, 10 tags
* Large menu: 100 posts, 100 tags

The database contained:
* 1238 post objects
* 15,000 term relationships

Anyway, I reckon this puppy does have some benefits and is probably worth putting in. 

These are the results for 1000 loops of each test case. 

| Menu Size | Cached | Patched time     | Patched Queries | Unpatched time  | Unpatched queries |
|:---------:|--------|------------------|-----------------|-----------------|-------------------|
| Small     | Y      | 1.0665791034698  | 1001            | 4.2827336788177 | 1005              |
| Small     | N      | 2.5907719135284  | 2000            | 10.757255315781 | 6000              |
| Med       | Y      | 1.2250320911407  | 1001            | 7.2486755847931 | 1005              |
| Med       | N      | 2.260603427887   | 2000            | 15.30788230896  | 6000              |
| Large     | Y      | 0.75003957748413 | 1001            | 54.079955101013 | 1005              |
| Large     | N      | 1.9239404201508  | 2000            | 89.751573801041 | 6000              |